modifiers/godot/voxel_modifier_gd.cpp

```diff
@@ -115,16 +85,55 @@ void VoxelModifier::_notification(int p_what) {
 				_modifier_id = 0;
 			}
 		} break;
+		case Node3D::NOTIFICATION_TRANSFORM_CHANGED: {
+			if (_volume == nullptr) {
+				Node *parent = get_parent();
+				VoxelLodTerrain *volume = Object::cast_to<VoxelLodTerrain>(parent);
+				// Using get_parent_node_3d when terrain is immediate parent
+				// would be a breaking change as it returns null if node is toplevel
+				if (volume == nullptr) {
+					Node3D *grandparent = get_parent_node_3d();
+					volume = Object::cast_to<VoxelLodTerrain>(grandparent);
+					while (grandparent != nullptr && volume == nullptr) {				
+						grandparent = grandparent->get_parent_node_3d();
+						volume = Object::cast_to<VoxelLodTerrain>(grandparent);
+					}
+				}
+				_volume = volume;
+			
+
+				if (_volume != nullptr) {
+					VoxelData &voxel_data = _volume->get_storage();
+					VoxelModifierStack &modifiers = voxel_data.get_modifiers();
+					const uint32_t id = modifiers.allocate_id();
+					zylann::voxel::VoxelModifier *modifier = create(modifiers, id);
+
+					if (modifier->is_sdf()) {
+						zylann::voxel::VoxelModifierSdf *sdf_modifier =
+								static_cast<zylann::voxel::VoxelModifierSdf *>(modifier);
+						sdf_modifier->set_operation(to_op(_operation));
+						sdf_modifier->set_smoothness(_smoothness);
+					}
+					// This transform is wrong, but will be updated with NOTIFICATION_TRANSFORM_CHANGED
+					// After node enters the tree
+					modifier->set_transform(get_transform());
+					_modifier_id = id;
+					// TODO Optimize: on loading of a scene, this could be very bad for performance because there could be,
+					// a lot of modifiers on the map, but there is no distinction possible in Godot at the moment...
+					post_edit_modifier(*_volume, modifier->get_aabb());
+				}
+			}
 
-		case Node3D::NOTIFICATION_LOCAL_TRANSFORM_CHANGED: {
+			update_configuration_warnings();
```

Comment:
Why should configuration warnings update when the transform changes?

Reply:
Ah, sorry. This was an artifact from copying the parenting code over to the transformation notification. Does it only need to be run if the modifier was added to the terrain?

Reply:
It may run only when the conditions checked are affected, so currently, only when `_volume` changes.